### PR TITLE
chore(yarn): update Yarn: 2.4.1 → 3.0.0

### PR DESCRIPTION
**Pull request by PiDelport**
Release post:

* https://dev.to/arcanis/yarn-3-0-performances-esbuild-better-patches-e07

Changes affecting this project:

* `.pnp.js` now called `.pnp.cjs`
* `yarn sdks` moved to package `@yarnpkg/sdks`

This release fixes the Yarn 2.4 performance issue where script execution became very slow:

> script execution tends to have a natural overhead, but bugs in 2.4 and prior caused this overhead to grow relative to the size of the project itself. This is no longer the case, and the overhead should now be constant."

Upgrade steps:

1. `yarn set version latest`
2. `yarn dlx @yarnpkg/sdks`
3. Replace all other references to `.pnp.js` with `.pnp.cjs`

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -19,4 +19,4 @@ plugins:
   - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
     spec: '@yarnpkg/plugin-interactive-tools'
```
Comment (merceyz) [inline .yarnrc.yml:20]: Should re-import this plugin to update it as well

```sh
yarn plugin import interactive-tools
```

File: Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ WORKDIR /app
 COPY ./package.json ./
 COPY ./dist ./dist
 COPY ./.yarn ./.yarn
-COPY ./.pnp.js ./
+COPY ./.pnp.cjs ./
 COPY ./yarn.lock ./
 EXPOSE 3000
 CMD [ "yarn", "start:prod" ]
```
Comment (merceyz) [inline Dockerfile:9]:
```suggestion
CMD [ "node", "-r", "./.pnp.cjs", "dist/main" ]
```

It's more efficient to start node directly

File: .yarnrc.yml (hunk 1 above)
Comment (PiDelport) [reply .yarnrc.yml:20]: Ah, thanks!

Comment (PiDelport) [reply .yarnrc.yml:20]: Applied in 2f47b81.

File: Dockerfile (hunk 2 above)
Comment (PiDelport) [reply Dockerfile:9]: More documentation on this: https://yarnpkg.com/features/pnp#initializing-pnp (We should probably link that from here, if we apply this change.)

@longtomjr, what do you think?

Comment (longtomjr) [reply Dockerfile:9]: Sounds good. If this reduce the overhead we should use node and link to the docs.

Comment (PiDelport) [reply Dockerfile:9]: I wouldn't think the overhead is a big concern in terms of performance (starting up the server is unlikely to become a bottleneck 😄), but avoiding the need to indirect through yarn is potentially a significant reduction in complexity for the deployable image, if we can omit yarn entirely, in favour of just including its output? We should maybe have a separate PR dedicated to that.

Comment (longtomjr) [reply Dockerfile:9]: Not sure if the cost is only for server startup, but I am happy either way. Does not seem like a massive change for this PR, and then we can have a backlog task / follow up pr to look at the deployed image complexity.

Comment (PiDelport) [reply Dockerfile:9]: It should only for startup: according to the docs, all `yarn` does is to add that flag before invoking `node`.

**Comment by PiDelport** [comment]
- Rolling this into #397

**Pull request closed without merging**